piccolo/testing/model_builder.py

```diff
@@ -0,0 +1,155 @@
+import asyncio
+import json
+import typing as t
+from datetime import date, datetime, time, timedelta
+from decimal import Decimal
+from uuid import UUID
+
+from piccolo.columns.base import Column
+from piccolo.table import Table
+from piccolo.testing.exceptions import InvalidColumnError
+from piccolo.testing.random_builder import RandomBuilder
+
+
+class ModelBuilder:
+    __DEFAULT_MAPPER: t.Dict[t.Type, t.Callable] = {
+        bool: RandomBuilder.next_bool,
+        bytes: RandomBuilder.next_bytes,
+        date: RandomBuilder.next_date,
+        datetime: RandomBuilder.next_datetime,
+        float: RandomBuilder.next_float,
+        int: RandomBuilder.next_int,
+        str: RandomBuilder.next_str,
+        time: RandomBuilder.next_time,
+        timedelta: RandomBuilder.next_timedelta,
+        UUID: RandomBuilder.next_uuid,
+    }
+
+    def __init__(self, minimal: bool = False, persist: bool = True):
+        """
+        Configurations for ModelBuilder
+
+        :param minimal:
+            If set, only required and non-nullable fields will be randomized.
+
+        :param persist:
+            If set, random model will not persisted by default.
+
+        """
+        self.minimal = minimal
+        self.persist = persist
+
+    async def build(self, table_class: t.Type[Table], **kwargs) -> Table:
+        """
+        Build Table instance with random data and save async.
+        This can build relationships, supported data types and parameters.
+
+        :param table_class:
+            Table class to randomize.
+
+        Examples:
+
+            manager = ModelBuilder.build(Manager)
+            manager = ModelBuilder.build(Manager, name='Guido')
+            manager = ModelBuilder(persist=False).build(Manager)
+            manager = ModelBuilder(minimal=True).build(Manager)
+            band = ModelBuilder.build(Band, manager=manager)
+        """
+        return await self._build(
+            table_class, _persist=self.persist, _minimal=self.minimal, **kwargs
+        )
+
+    def build_sync(self, table_class: t.Type[Table], **kwargs) -> Table:
+        """
+        Build Table instance with random data and save sync.
+        This can build relationships, supported data types and parameters.
+
+        :param table_class:
+            Table class to randomize.
+
+        Examples:
+
+            manager = ModelBuilder.build_sync(Manager)
+            manager = ModelBuilder.build_sync(Manager, name='Guido')
+            manager = ModelBuilder(persist=False).build_sync(Manager)
+            manager = ModelBuilder(minimal=True).build_sync(Manager)
+            band = ModelBuilder.build_sync(Band, manager=manager)
+        """
+        return asyncio.run(
```

Comment:
Can use `run_sync` from `piccolo.utils.sync` here. It's pretty similar, but slightly more robust.